app/cdap/components/common/InlayDrawerWidget/__tests__/InlayDrawerWidget.test.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { cleanup, fireEvent, render } from '@testing-library/react';
+import InlayDrawerWidget from 'components/common/InlayDrawerWidget';
+import { IActionsOptions } from 'components/common/InlayDrawerWidget/Menu';
+import React from 'react';
+
+const handleDrawerCloseIconClick = jest.fn();
+const onSaveButtonClick = jest.fn();
+const onApplyButtonClick = jest.fn();
+const onDownloadClick = jest.fn();
+
+const actionsOptions: IActionsOptions[] = [
```

Comment:
most of these unit tests are not necessary if you use the material ui menu component.

Reply:
these unit tests are testing the behavior by checking whether the button clicks will trigger the appropriate event in the parent. as this is a common component, I think this should be sufficient.
please let me know if you see I have missed any test cases.